tests/network/services.go

```diff
@@ -47,6 +49,34 @@ var _ = SIGDescribe("Services", func() {
 		return job
 	}
 
+	exposeExistingVMISpec := func(vmi *v1.VirtualMachineInstance, subdomain string, hostname string, selectorLabelKey string, selectorLabelValue string) *v1.VirtualMachineInstance {
```

Comment:
So I can re-use it ?... 

Please re-scope your question, I'm afraid I don't understand what is your doubt.

Reply:
Of course you have it to reuse it.. I mean why do you need to set `Hostname`, `Subdomain` and add label a used will choose?

Reply:
:) right.

`Hostname` & `Subdomain` are used in [this test](https://github.com/kubevirt/kubevirt/blob/master/tests/vmi_networking_test.go#L331). It will be how the traffic is routed into the VMI. Check [headless services k8s docs](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) for more info.

The label key / value are how the service will be 'mapped' to the VM, check the [k8s service docs](https://kubernetes.io/docs/concepts/services-networking/service/#defining-a-service).

Reply:
This is something local to this test. I'd prefer to keep the scope of this short, at least until this is required in another module.

Reply:
Ok, but note that the advantage is that with With*, you do it on creation and all details are presented to you.